autogtp/Job.cpp

```diff
@@ -158,25 +161,27 @@ Result ValidationJob::execute(){
     QString wmove = "play white ";
     QString bmove = "play black ";
     do {
-        first.move();
-        if (!first.waitForMove()) {
-            return res;
-        }
-        first.readMove();
-       m_boss->incMoves();
-        if (first.checkGameEnd()) {
-            break;
+        if (first.getToMove() == Game::BLACK) {
+            first.move();
+            if (!first.waitForMove()) {
+                return res;
+            }
+            first.readMove();
+            m_boss->incMoves();
+            if (first.checkGameEnd()) {
+                break;
+            }
+            second.setMove(bmove + first.getMove());
+            first.nextMove();
```

Comment:
I would put the if outside the do ... while loop. Something like that:

     if (first.getToMove() == Game::White) {
          second.move();
          if (!second.waitForMove()) {
              return res;
          }
          second.readMove();
          m_boss->incMoves();
          first.setMove(wmove + second.getMove());
    }
    do {
    ...
    } while (second.nextMove() && m_state.load() == RUNNING);

Reply:
I agree that will be easier to read. Will change.